ngraph/core/src/op/log_softmax.cpp

```diff
@@ -0,0 +1,111 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/log_softmax.hpp"
+
+#include "itt.hpp"
+#include "ngraph/runtime/reference/log_softmax.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v5::LogSoftmax, "LogSoftmax", 5);
+
+op::v5::LogSoftmax::LogSoftmax(const Output<Node>& arg, const int64_t axis)
+    : Op({arg})
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v5::LogSoftmax::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("axis", m_axis);
+    return true;
+}
+
+void op::v5::LogSoftmax::validate_and_infer_types()
+{
+    const PartialShape& input_shape = get_input_partial_shape(0);
+    if (input_shape.rank().is_static())
+        NODE_VALIDATION_CHECK(this,
+                              m_axis < input_shape.rank().get_length() &&
+                                  m_axis >= -input_shape.rank().get_length(),
+                              "Reduction axis (",
+                              m_axis,
+                              ") is out of bounds (argument shape: ",
+                              input_shape,
+                              ").");
+
+    set_output_type(0, get_input_element_type(0), input_shape);
+}
+
+shared_ptr<Node> op::v5::LogSoftmax::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<op::v5::LogSoftmax>(new_args.at(0), m_axis);
+}
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool try_evaluate_log_softmax(const HostTensorPtr& arg,
+                                         const HostTensorPtr& out,
+                                         const Shape& shape,
+                                         const AxisSet& axes)
+    {
+        return (ET == arg->get_element_type()) &&
+               (runtime::reference::log_softmax(
+                    arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), shape, axes),
+                true);
+    }
+
+    bool evaluate_log_softmax(const HostTensorPtr& arg,
+                              const HostTensorPtr& out,
+                              const AxisSet& axes)
+    {
+        auto shape = out->get_shape();
+        return try_evaluate_log_softmax<element::Type_t::f16>(arg, out, shape, axes) ||
+               try_evaluate_log_softmax<element::Type_t::f32>(arg, out, shape, axes) ||
+               try_evaluate_log_softmax<element::Type_t::f64>(arg, out, shape, axes);
+    }
+} // namespace
+
+bool op::v5::LogSoftmax::evaluate(const HostTensorVector& outputs,
+                                  const HostTensorVector& inputs) const
+{
+    OV_ITT_SCOPED_TASK(itt::domains::nGraphOp, "op::v5::LogSoftmax::evaluate");
+    outputs[0]->set_unary(inputs[0]);
+    size_t axis;
```

Comment:
`size_t` -> `int64_t`

Reply:
No, this is right. This variable is non-negative.